### DISCUSSION TITLE
Din Rail Mini Wifi Circuit Breaker

### DISCUSSION THREAD

**Discussion by pirracas77** [arendst/Tasmota#10915] [Q&A] [answered]
Does anyone this device could be tasmotized?

https://www.aliexpress.com/item/4000490400643.html

I have tried via sonOTA without sucess.

**Comment by sfromis** [comment, marked as answer]
SonOTA is a long dead project, basically a waste of time to try. eWelink is not a good indicator of having a supportable ESP8266 inside, as that firmware also runs on at least one other chip type. If you have the device at hand, you can find the MAC address, and look it up.

[Tasmota Device Templates Repository](https://templates.blakadder.com/) has devices known to have been supported, at least at one point in time. Outside of that, the general answer is "we don't know till someone finds out, and reports the result to the templates repository".

**Reply by pirracas77** [reply]
Thanks for you quick answer.

What is supposed that I have to do with the MAC? What part of the MAC address will help me to find something ?

**Reply by barbudor** [reply]
Use the whole MAC address on one of the numerous sites that state ["MAC address identification"](https://www.google.com/search?q=mac+address+identification) and that will let you know the chip vendor's name. Espressiff would be the one
Sometime manufacturers use their own pool of MAC address in that case that will not give info
So openning is generally the most secure way of knowing for sure

**Reply by pirracas77** [reply]
oh, great!

**Reply by pirracas77** [reply]
It seems the chip is OK.

`
Address Prefix | C4:4F:33
-- | --
Vendor / Company | Espressif Inc.
Start Address | C44F33000000
End Address | C44F33FFFFFF
Company Address | Room 204, Building 2, 690 Bibo Rd, Pudong New Area Shanghai Shanghai 201203 Cn
`


Unfortunatelly I cannot open it without damage it.

Any advice about next steps?

**Reply by barbudor** [reply]
Serial flashing requires opening unfortunately
I have no experience with those specific devices.
Unfortunately sometime you have to destroy one to understand how it is made and find a better opening way for the next ones.

**Comment by pirracas77** [comment]
Well, finally I decided to open it. It seems there is no any pin RX,TX or something like this.


![1613060568322](https://user-images.githubusercontent.com/18216098/107666405-c7500c80-6c8e-11eb-803d-af9eb5209bef.jpg)
![1613060568347](https://user-images.githubusercontent.com/18216098/107666412-c8813980-6c8e-11eb-8e11-2fb346f39afb.jpg)
![1613060568352](https://user-images.githubusercontent.com/18216098/107666413-c919d000-6c8e-11eb-98fb-ea0b8d109a83.jpg)

**Reply by barbudor** [reply]
pinout of the module is there https://tasmota.github.io/docs/Pinouts/#psf-b85psf-b01psf-b04

**Reply by pirracas77** [reply]
Perfect!
I will try this weekend.

**Comment by pirracas77** [comment]
Well, finally I got it.


![1618762533273](https://user-images.githubusercontent.com/18216098/115152804-032e9600-a073-11eb-8c35-2a884efa397b.jpg)
![1618762533258](https://user-images.githubusercontent.com/18216098/115152801-01fd6900-a073-11eb-8d21-90e049de0f22.jpg)

The 3.3v pin and GND are available in the board (blue and red cables in the picture)
It enters in flashing mode keeping the button pressed while you connect USB-TTL adapter.

**Comment by blakadder** [comment]
The submitted template for this devices was just a generic module

**Reply by pirracas77** [reply]
Yes sorry, no idea how to improve it.

**Reply by blakadder** [reply]
Just paste the correct template here

**Reply by pirracas77** [reply]
What I try to explain is that I don't know how to improve the template that I wrote. No idea what is the process to discover its own template. The sonoff basic template just works.

**Reply by blakadder** [reply]
Gotcha!

**Comment by falconhunter80** [comment]
Hi,
Which template configuration (pin-out) did you use?

**Reply by barbudor** [reply]
He wrote "Sonoff Basic" so that would be this one: https://templates.blakadder.com/sonoff_basic.html

**Reply by pirracas77** [reply]
> Hi, Which template configuration (pin-out) did you use?

This one
https://templates.blakadder.com/sohan_RDCBC-1P.html

**Reply by sfromis** [reply]
Actually same pins as the Sonoff Basic, both should work the same.

**Reply by falconhunter80** [reply]
Thanks for your quick answer.
I have two types of circuit breaker although when I disassembled them I saw that the hardware matches the one shown by @pirracas77.
I have used the mentioned template and it is already working.
The references I have are OPCBC-1P and ZNWF-1P, which are the same hardware as RDCBC-1P but with different branding.
![IMG20220531130618_low](https://user-images.githubusercontent.com/67373979/171161159-309a29bd-2fcf-4fe1-8f57-72fe0492d817.jpg)
I am not sure if I must add two new templates or submit any indication in the existing one to the [Tasmota Device Templates Repository](https://templates.blakadder.com/) .

**Reply by eteixido1** [reply]
> Thanks for your quick answer. I have two types of circuit breaker although when I disassembled them I saw that the hardware matches the one shown by @pirracas77. I have used the mentioned template and it is already working. The references I have are OPCBC-1P and ZNWF-1P, which are the same hardware as RDCBC-1P but with different branding. ![IMG20220531130618_low](https://user-images.githubusercontent.com/67373979/171161159-309a29bd-2fcf-4fe1-8f57-72fe0492d817.jpg) I am not sure if I must add two new templates or submit any indication in the existing one to the [Tasmota Device Templates Repository](https://templates.blakadder.com/) .

Hello @falconhunter80, I'm looking for this kind of breakers but EARU, TOMZN, SOHAN and other use the same bolts (TONGOU with meter solution). Could you please let us known if OPCBC-1P or ZNWF-1P use different screws? I'm really worried about assurance company if something happens and they find a modified breaker in fusebox.

Kind regards,

**Reply by falconhunter80** [reply]
Hi @eteixido1 

OPCBC-1P and ZNWF-1P use same bolts. I removed them by drilling head, pushing with a punch and pulling with pliers.

![IMG20220531080053_2](https://user-images.githubusercontent.com/67373979/188552461-04d9ce4e-9c32-47d9-8048-d776207cdd3e.jpg)

I have placed DIN 84 M2.5 x 10 screws threaded in the same bolt holes plastic and they fit very well. 
Sorry, I do not have any photo with final result, but it results more "professional" than fixing with insulating tape.

KR

**Reply by pirracas77** [reply]
very good idea. I take note about the screws.

**Reply by tokamac** [reply]
These are not "bolts" or "screws", these are _rivets_. Meaning that they must be crushed and destroyed (as shown in the picture just above) to remove them and open the circuit breaker, which can no longer be closed properly (look above with the one covered with blue tape!). Too messy, not a satisfactory solution. Are there Wi-Fi DIN rail circuit breakers out there with **regular screws** that can be disassembled in a clean way?

At this point I conclude two things skimming through the Tasmota Supported Devices Repository:
- Only the black TOMZ DIN rail switches (these are not MCBs or RCCBs) casing seem to have Phillips screws according to the notes there. The TOMZ white casing seems to have plastic clips with no glue. Others have rivets. If ever the notes there are reliable.
- There are other brands that have the same casing as TOMZ (for example EARU) but most of them use Tuya instead of eWeLink, increasing the risk of having a non-compatible chip.

As for me, I am searching for 2P switches (that switch both neutral and live wires, not the thinner 1P version that switches only the live wire). So far I found two possible references (not tested yet): 
- TOMZ [TOWBC-2P-e](https://www.aliexpress.com/item/4001125085099.html) (2P, single-phase, 80A, eWeLink) 
- TONGU [AT-Q-SR263WE](https://www.aliexpress.com/item/1005003001345072.html) (2P, single-phase, 63A, integrated 30mA RCD, eWeLink). Here is the [product page](https://elcb.net/product/wifi-circuit-breaker-switch-remote-control/)

Has anyone tested others without rivets?

**Reply by barbudor** [reply]
The Tongu doesn't seem to have screews neither.
I can see the same unit cheaper at EARU: https://www.aliexpress.com/item/1005006026136272.html (not ordered)

I also found this one: https://www.aliexpress.com/item/4001220603015.html
The diagram seems to say that the N is also switchable
I'm ordering one to see

But I fear you are right that's is not possible to find a P+N switch within a single width module.

**Reply by juskalalie** [reply]
Hi Barbudor,
I understand you ordered this relay: https://www.aliexpress.com/item/4001220603015.html
If yes, it would be great to know your results. I am also looking for a good circuit breaker for at least 20A that can be flashed with tasmota...
Cheers!

**Reply by barbudor** [reply]
Hi
Actually I bought this RDCP-1P and a EACKB-EWE-M (including power meter)

I have received the 2nd one, flashed tasmota and configured.
Flashing was a bit tedious as the power metering chip is connected to the serial RX and was so conflicting with the serial flasher. I had to un solder and gently lift the pin of the metering chip to perform the flashing.
I now have 1 led, relay and power meter working.
Still I see that there are connections to a GPIO that I haven't identified as well as the ADC. The ADC seems connected to a current transformer which I don't get the usage as there is already a CSE7758 as power meter. May be that for security as it is supposed to be used as a breaker.
The relay is only on Line as given for 63A.
I will post some photos and template later.
Also I still have to find a proper way of closing it back as I have to cut the rivets. 

The RDCP is expected this week so I'll likely work on it next week end.

**Reply by tokamac** [reply]
> Actually I bought this **RDCP-1P** 

I think you meant **RDCBC-1P**?

> and a **EACKB-EWE-M** (including power meter)
I have received the 2nd one, flashed tasmota and configured.
Flashing was a bit tedious as the power metering chip is connected to the serial RX and was so conflicting with the serial flasher. I had to un solder and gently lift the pin of the metering chip to perform the flashing.

The EARU **EACKB-EWE-M** is the eWeLink "multifunction" version, i.e. with the power meter function. It also exists in another variant, still with eWeLink but as "single function" (i.e. without the chip that gave you trouble): [EARU **EAWCBE**](https://www.aliexpress.com/item/1005006262320403.html). I think the "EweLink single version" (without the power meter chip) is preferable to choose for other people wanting to flash these with Tasmota.

**Reply by barbudor** [reply]
Strange reasonning
If you want the power meter, you should buy one with power metering
As I wrote Tasmota works fine on the multi version

**Reply by sfromis** [reply]
Looks like the point was it being more trouble to flash the one with energy monitoring due to that chip interfering with flashing. Hence life will be easier for "other people" not really needing this feature.

**Reply by juskalalie** [reply]
Sounds promising! I'd love to have power metering feature too, but mainly I need a 20A circuit breaker that can be controlled (on/off) via WiFi. Maybe I'll give EACKB-EWE-M a try! :)

**Reply by tokamac** [reply]
> Strange reasonning
> If you want the power meter, you should buy one with power metering

Of course. As for me, I am searching for a Wi-Fi contactor (2P) with no power metering, and my message was specifically aimed at those who want the simplest seamless procedure (quote: "for other people…") especially considering soldering. It was not intended specifically to you if you absolutely want power metering. Plus, this thread is called "Din Rail Mini Wifi Circuit Breaker" not "Din Rail Mini Wifi Circuit Breaker with Power Meatering", so my reasoning was not "strange" at all, you simply didn't try to grasp the broader context.

> Looks like the point was it being more trouble to flash the one with energy monitoring due to that chip interfering with flashing. Hence life will be easier for "other people" not really needing this feature.

My point exactly. Thanks.

**Comment by laurentppol** [comment]
[OT]
Did no one read marking on case and on "product" page?
"50A"? On < 0.5mm2 wire?
Nice method to have a BIG boooom in fusebox.
I was tought that we use 1mm2 for 10A, for Cu (for cables, for short traces maybe less).

[who can tell now, that Chineses don't write SF ;p ]

**Comment by lalo-uy** [comment]
The wires are just for the relay coil.
The real conductores are metal bars you see on the other side of the case.

**Comment by paulius096** [comment]
Hi is it possible to expand wifi range? Or use some wifi antenna to increase wifi range? i can not see any antenna in photos

**Reply by sfromis** [reply]
The antenna is inside the plastic shell, a small blue ceramic antenna at the corner of the PSF-B01 module. A major factor to signal quality is how much metal is around the breaker panel. Like when a metal door can block/reflect a lot of radio waves, and metal parts near the breaker can also make a lot of difference.

**Comment by tokamac** [comment]
It ordered an **EARU RDCBC-2P** DIN relay: [https://www.aliexpress.com/item/1005004832366175.html](https://www.aliexpress.com/item/1005004832366175.html)
Options: _Single Phase 2_, _63A_, _WiFI eWeLink (White)_

Several remarks on the product:
* Pay attention to the color as the black version (Tuya) is said to have a newer chip that is incompatible with Tasmota. The chip inside the white version (eWeLink) is PSF-B01, based on ESP8285.
*  You can order the smaller 1P (32A or 50A) versions, but the enclosure of this one is secured with rivets as shown above by several posters, which you'll need to drill and destroy in order to be able to open the case, and it won't closed properly again afterward. On the contrary, the 2P and 4P enclosures are closed with just two screws on the back. Simple to open and close as you wish. 63A or 80A, il doesn't matter.
* 1P circuit breakers cut the live only, while 2P are supposed to cut both live and neutral. But opening the 2P enclosure reveals that the relay only cuts the live, exactly like the 1P version. Neutral wire is always active. However, the Chinese clearly state this on the product page, they don't lie. So in the end, the larger 2P module does exactly the same thing as the 1P module, except that it is rated at a higher maximum current, and has no rivets.

The most tedious part was to solder thin wires directly to the tiny pins on the chip, very close to each other (there is no larger pins or holes available on the PCB) – I am still wondering how I eventually managed to do it properly. 

The goal is to control the outlet of a powerful hot-water tank heater.

![IMG_6736](https://github.com/arendst/Tasmota/assets/3685009/6e3db685-e74e-46f0-a026-20ccf88377ba)
![IMG_6737](https://github.com/arendst/Tasmota/assets/3685009/fb07937b-3bf3-4f14-93ba-d459697c3d6e)
![IMG_6739](https://github.com/arendst/Tasmota/assets/3685009/e4b4f84f-7586-4963-8bb2-3fcc31b7331f)
![IMG_6740](https://github.com/arendst/Tasmota/assets/3685009/ff9fbec5-3356-42b9-9596-b867cb8b12cc)
![IMG_6741](https://github.com/arendst/Tasmota/assets/3685009/26d94362-8b12-4da9-899c-e599524483c6)
![IMG_6738](https://github.com/arendst/Tasmota/assets/3685009/bafb2010-bf2f-4ec3-8acb-32ec94ba9130)
![IMG_6742](https://github.com/arendst/Tasmota/assets/3685009/545466f5-b9ff-44ee-a32e-450e9429faea)
![IMG_6743](https://github.com/arendst/Tasmota/assets/3685009/3bc689d6-410a-4b55-9c5f-66aaf07307fc)
![IMG_6744](https://github.com/arendst/Tasmota/assets/3685009/7248d329-b9e7-4793-95c7-22ce03eba20c)
